### PR TITLE
[AssetMapper] Split ImportmapManager into 2

### PR DESCRIPTION
**Pull request by weaverryan**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | yes? (performance bug)
| New feature?  | yes
| Deprecations? | no
| Tickets       | None
| License       | MIT

Hi!

Just a simple "copy-and-paste" of methods from `ImportMapManager` to a new `ImportMapGenerator`. The motivation is performance: `ImportMapGenerator` is instantiated on production on every request, so this PR results in 2 unnecessary services (`RemotePackageDownloader` and `JsDelivrEsmResolver`) no longer being instantiated.

Thanks!

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/asset_mapper.php [hunk 1]
```diff
@@ -198,7 +205,6 @@
         ->set('asset_mapper.importmap.command.require', ImportMapRequireCommand::class)
             ->args([
                 service('asset_mapper.importmap.manager'),
-                param('kernel.project_dir'),
```
Comment (weaverryan) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/asset_mapper.php:208]: Argument wasn't used - unrelated to this PR

File: src/Symfony/Component/AssetMapper/ImportMap/ImportMapGenerator.php [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+
+use Symfony\Component\AssetMapper\AssetMapperInterface;
+use Symfony\Component\AssetMapper\Exception\LogicException;
+use Symfony\Component\AssetMapper\MappedAsset;
+use Symfony\Component\AssetMapper\Path\PublicAssetsPathResolverInterface;
+
+/**
+ * Provides data needed to write the importmap & preloads.
+ */
+class ImportMapGenerator
```
Comment (weaverryan) [inline src/Symfony/Component/AssetMapper/ImportMap/ImportMapGenerator.php:22]: Class is a copy-and-paste of methods from `ImportMapManager` with very few changes

**Review by stof — APPROVED** [review]

**Comment by chalasr** [comment]
The changelog should probably mention this change

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @weaverryan.

**Pull request merged**